airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/TeardownTask.kt

```diff
@@ -17,14 +20,27 @@ interface TeardownTask : SyncTask
  * TODO: Report teardown-complete and let the task launcher decide what to do next.
  */
 class DefaultTeardownTask(
+    private val checkpointManager: CheckpointManager<*, *>,
+    private val syncManager: SyncManager,
     private val destination: DestinationWriter,
     private val taskLauncher: DestinationTaskLauncher
 ) : TeardownTask {
     val log = KotlinLogging.logger {}
 
+    private val teardownHasRun = AtomicBoolean(false)
+
     override suspend fun execute() {
-        destination.teardown()
-        taskLauncher.handleTeardownComplete()
+        // TODO: This should be its own task, dispatched on a timer or something.
```

Comment:
nonblocking: why a timer? shouldn't we just call this any time we get a PERSISTED batch?

Reply:
Yes, that's the best place for it! We also later need a timed task to try to force a flush every 5 minutes, but that's a whole other thing.